src/simulation_models/stage_update.jl

```diff
@@ -185,6 +208,8 @@ end
 #############################Interfacing Functions##########################################
 
 function cache_update!(stage::_Stage)
+
```

Comment:
Why did you add two blank lines?  As far as I can tell the Julia convention, like any other language, is it have no blank lines in between the function declaration and the first line.